crates/staged-sync/src/utils/versioning/client.rs

```diff
@@ -0,0 +1,157 @@
+use std::{
+    fs, io,
+    io::Write,
+    path::{Path, PathBuf},
+    time::{SystemTime, UNIX_EPOCH},
+};
+use tracing::info;
+
+/// The name of the file that contains the version of the client.
+const CLIENT_VERSION_FILE_NAME: &str = "client.version";
+/// The version of the client stored in the [CLIENT_VERSION_FILE_NAME] file in the same directory as
+/// database. Example: `0.1.0-e43455c2`
+pub(crate) const CLIENT_VERSION: &str =
+    concat!(env!("CARGO_PKG_VERSION"), "-", env!("VERGEN_GIT_SHA"));
```

Comment:
What is this used for? Dont we already have the CLI version?

Reply:
The CLI version is a human-readable string that's used for `--version` and printed to logs on startup. This version with current timestamp is stored in the file, which is updated every time the node is updated. The idea is that this version history may be useful for debugging when something goes wrong after the update.